lib/puppet/provider/user/useradd.rb

```diff
@@ -9,8 +9,12 @@
     install Ruby's shadow password library (often known as `ruby-libshadow`)
     if you wish to manage user passwords."
 
+  # These commands have special names for use with the base ObjectAdd provider
   commands :add => "useradd", :delete => "userdel", :modify => "usermod", :password => "chage"
 
+  # This is used in our local implementation and has no special name
+  commands :gpasswd => "gpasswd"
```

Comment:
For the exact same reason as ticket 1942 - the user in the case of the ticket (PUP-3774) is coming from AD, so doesn't have an entry in /etc/passwd. We need to use gpasswd in this case.

I can expand the commit message with that note.